src/main.c

```diff
@@ -24,17 +16,18 @@ int main()
 
     sch_init_t1();
     sch_start();
-
-    if (SENSOR_MODE == TEMPERATURE_SENSOR)
-    {
-        sch_add_task(run_temperature_scan, 4000, 4000);
-    }
-    else
-    {
-        sch_add_task(run_light_scan, 3000, 3000);
-    }
-
-    sch_add_task(SENSOR_MODE == TEMPERATURE_SENSOR ? report_average_temperature : report_average_temperature, 0, 6000);
+    
+	if (SENSOR_MODE == 0) // Light sensor
```

Comment:
Can be simplified to
``int time = SENSOR_MODE == 0 ? 3000 : 4000;
sch_add_task(run_sensor_scan, time, time);``